lint.go

```diff
@@ -1394,6 +1396,59 @@ func (f *file) lintTimeNames() {
 	})
 }
 
+func (f *file) lintDurationUnits() {
+	f.walk(func(node ast.Node) bool {
+		e, ok := node.(ast.Expr)
+		if !ok {
+			return true
+		}
+
+		// Don't complain about time.Duration(30), at least for now.
+		if call, ok := node.(*ast.CallExpr); ok {
+			if len(call.Args) == 1 {
+				sel, ok := call.Fun.(*ast.SelectorExpr)
+				if ok && sel.Sel.Name == "Duration" {
+					if ident, ok := sel.X.(*ast.Ident); ok && ident.Name == "time" {
+						return false
+					}
+				}
+			}
+		}
+
+		tv := f.pkg.typesInfo.Types[e]
+		if tv.Type == nil || !f.pkg.isNamedType(tv.Type, "time", "Duration") {
+			return true
+		}
+		if tv.Value == nil {
+			return true
+		}
+		// Ok for 0 to be unitless; non-constants are ok
+		if x, ok := exact.Int64Val(tv.Value); !ok || x == 0 {
+			return true
+		}
+		if _, ok := e.(*ast.BasicLit); !ok {
+			// Don't descend into complex expressions
+			return false
+		}
+
+		f.errorf(e, 0.8, category("time"), "time.Duration without unit: %v", f.render(e))
```

Comment:
I think the message needs to be something like "time.Duration expression should use time package constants".